front/app/containers/CustomPageShow/index.tsx

```diff
@@ -12,38 +16,131 @@ import { useParams } from 'react-router-dom';
+`;
+
+export const PageContent = styled.main`
+  flex-shrink: 0;
+  flex-grow: 1;
+  background: #fff;
+  padding-top: 60px;
+  padding-bottom: 100px;
+`;
+
+export const PageTitle = styled.h1`
+  color: ${({ theme }) => theme.colors.tenantText};
+  font-size: ${fontSizes.xxxxl}px;
+  line-height: normal;
+  font-weight: 600;
+  text-align: left;
+  margin: 0;
+  padding: 0;
+  padding-top: 0px;
+  padding-bottom: 40px;
+
+  ${media.tablet`
+    font-size: ${fontSizes.xxxl};
+  `}
+  ${isRtl`
+    text-align: right;
+    direction: rtl;
+  `}
+`;
+
+const CustomPageShow = ({ intl: { formatMessage } }: InjectedIntlProps) => {
   const { slug } = useParams() as {
     slug: string;
   };
-  const customPage = usePage({ pageSlug: slug });
+  const page = usePage({ pageSlug: slug });
+  const localize = useLocalize();
 
-  if (isNilOrError(customPage)) {
+  if (isNilOrError(page)) {
     return null;
   }
 
-  const attributes = customPage.attributes as unknown as ICustomPagesAttributes;
+  let seoTitle: string;
+  let seoDescription: string;
+  let blockIndexing: boolean;
+  let pageTitle: string;
+  let pageDescription: JSX.Element;
+  let pageSlug: string;
+
+  if (!isNilOrError(page)) {
+    seoTitle = localize(page.attributes.title_multiloc);
+    seoDescription = '';
+    blockIndexing = false;
+    pageSlug = page.attributes.slug || '';
+    pageTitle = localize(page.attributes.title_multiloc);
+    pageDescription = (
+      <ResolveTextVariables value={page.attributes.top_info_section_multiloc}>
+        {(multiloc) => <T value={multiloc} supportHtml={true} />}
+      </ResolveTextVariables>
+    );
+  } else {
+    seoTitle = formatMessage(messages.notFoundTitle);
+    seoDescription = formatMessage(messages.notFoundDescription);
+    blockIndexing = true;
+    pageTitle = formatMessage(messages.notFoundTitle);
+    pageDescription = <FormattedMessage {...messages.notFoundDescription} />;
+    pageSlug = '';
+  }
+
+  const attributes = page.attributes as unknown as ICustomPageAttributes;
   return (
-    <>
-      <Container id="e2e-landing-page">
-        <CustomPageHeader
-          headerLayout={attributes.banner_layout}
-          header_bg={attributes.header_bg}
-          headerMultiloc={attributes.banner_header_multiloc}
-          subheaderMultiloc={attributes.banner_subheader_multiloc}
-          headerColor={attributes.banner_overlay_color}
-          headerOpacity={attributes.banner_overlay_opacity}
-        />
-        <Content>
-          <TopInfoSection
-            multilocContent={attributes.top_info_section_multiloc}
-          />
-        </Content>
-      </Container>
-    </>
+    <Container className={`e2e-page-${pageSlug}`}>
+      <Helmet>
+        <title>{seoTitle}</title>
+        <meta name="description" content={seoDescription} />
+        {blockIndexing && <meta name="robots" content="noindex" />}
+      </Helmet>
+      <PageContent>
+        <StyledContentContainer>
+          <Fragment
```

Comment:
The `fragment` was (as far as I know) a hack to let additional HTML be inserted into homepages, we don't need it in custom pages

Reply:
Makes sense 👍